backend/src/apiserver/client/argo_fake.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	argoprojv1alpha1 "github.com/argoproj/argo/pkg/client/clientset/versioned/typed/workflow/v1alpha1"
+)
+
+type FakeArgoClient struct {
+	workflowClientFake FakeWorkflowClientInterface
+}
+
+func NewFakeArgoClient() *FakeArgoClient {
+	return &FakeArgoClient{NewWorkflowClientFake()}
+}
+
+func (c *FakeArgoClient) Workflow(namespace string) argoprojv1alpha1.WorkflowInterface {
```

Comment:
what is the difference between Workflow() and FakeWorkflow() and when should i use what?

Reply:
FakeWorkflow() is only used when users need to call the func in FakeWorkflowClientInterface.
The current problems is Workflow() needs to be implemented for the ArgoClientInterface while the resource_manager_test uses the func in FakeWorkflowClientInterface, such as GetWorkflowCount, GetWorkflowKeys

Reply:
Removed.